src/Paket.Core/Installation/InstallProcess.fs

```diff
@@ -390,73 +390,79 @@ let InstallIntoProjects(options : InstallerOptions, forceTouch, dependenciesFile
             d, Seq.append errorMessages groupErrors
 
         let usedPackages, errorMessages =
-            let dict = System.Collections.Generic.Dictionary<PackageName,SemVerInfo>()
+            let dict = System.Collections.Generic.Dictionary<PackageName,SemVerInfo*bool>()
             let errors = ResizeArray ()
             usedPackages
-            |> Map.filter (fun (_groupName,packageName) (v,_) -> 
+            |> Map.filter (fun (_groupName,packageName) (v,model) ->
+                let hasCondition = model.ReferenceCondition.IsSome
                 match dict.TryGetValue packageName with
-                | true,v' -> 
-                    if v' = v then false else
-                    errors.Add <| sprintf "Package %O is referenced in different versions in %s (%O vs %O), check the lock-file." packageName project.FileName v' v
-                    false
+                | true,(v',true) when hasCondition ->
+                    true
+                | true,(v',hasCondition') ->
+                    if v' = v then
+                        traceWarnfn "Package %O is referenced through multiple groups in %s (inspect lockfile for details). To resolve this warning use a single group for this project to get a unified dependency resolution or use conditions on the groups if you know what you are doing." packageName project.FileName
+                        false
```

Comment:
/cc @agross This basically means you get this warning on your example from https://github.com/fsprojects/Paket/issues/1458. IMHO we should not encourage this kind of paket group usage.